// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.uquantapi";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";
//option cc_generic_services = false;

package uquantapi;  // namespace

// The greeting service definition.
service Greeter {  //class
		// Sends a greeting
		rpc SayHello (HelloRequest) returns (HelloReply) {}
		rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
		rpc getFutureBars (kBarRequest) returns (kBarReply) {}
}

service FutureApi {
		rpc getFutureBars (kBarRequest) returns (kBarReply) {}
}

service StockApi {
		rpc getFutureBars (kBarRequest) returns (kBarReply) {}
}

// The request message containing the user's name.
message HelloRequest {
		string name = 1;
}

// The response message containing the greetings
message HelloReply {
		string message = 1;
}

message kBarRequest {
		string id = 1;
		string start = 2;
		string end = 3;
		int32  frequency = 4;
}

message kBar {
		string b =1;
		string e =2;
		double o =3;
		double h =4;
		double l =5;
		double c =6;
}

message kBarReply {
		repeated kBar kk = 1;
}
